### PR TITLE
Direct/GatewayConnectionConfigChange

### PR DESCRIPTION
**Pull request by xinlian12**
For issue: https://github.com/Azure/azure-sdk-for-java/issues/11531

1. Hide requestTimeout from DirectConnectionConfig and GatewayConnectionConfig
2. getConnectionTimeout -> getConnectTimeout
    setConnectionTimeout -> setConnectTimeout in DirectConnectionConfig

### CODE REVIEW

**Comment by xinlian12** [comment]
There is another proposed change is to initialize TimeoutHelper in ReplicatedResourceClient with constant value 30s.  But seems like in ReplicatedResourceClient, requestTimeout never being used. 

For request happened the first time, currently the service is using 60s which coming from
  InitialArgumentValuePolicyArg = Quadruple.with(false, false,Duration.ofSeconds(60), 0);

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/DirectConnectionConfig.java [hunk 1]
```diff
@@ -201,33 +201,32 @@ public DirectConnectionConfig setMaxRequestsPerConnection(int maxRequestsPerConn
...
      * This represents the timeout interval for requests
      *
-     * Default value is 60 seconds
+     * Default value is 5 seconds
      *
      * @param requestTimeout the request timeout interval
      * @return the {@link DirectConnectionConfig}
      */
-    public DirectConnectionConfig setRequestTimeout(Duration requestTimeout) {
+    DirectConnectionConfig setRequestTimeout(Duration requestTimeout) {
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/DirectConnectionConfig.java:217]: does consistency stack use this timeout?

Comment (xinlian12) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/DirectConnectionConfig.java:217]: From my current understanding. the consistency stack is not using it. 

For the request happens the first time, it is using 60s which comes from InitialArgumentValuePolicyArg = Quadruple.with(false, false,Duration.ofSeconds(60), 0);

**Comment by kushagraThapar** [comment]
@moderakh @xinlian12 - with this PR - we are keeping the consistency stack timeout to default value of 60 seconds initially as defined in `BackoffRetryUtility`. 30 seconds in case of all consistency except Strong and 60 seconds in case of Strong consistency.  
I think that sounds good to me.

**Comment by liquidvapour** [comment]
Hey @xinlian12 

Sorry for coming to the party a bit late but... Why was setRequestTimeout made private? What do we do if we want a timeout shorter than the default (5 seconds)?

The issue (#11531) does not really give any context for this change either.

**Comment by adelamodwala** [comment]
Agreed with @liquidvapour - why did this change need to happen @xinlian12 ? I would like to be able to define a shorter request timeout if possible.

**Comment by okohub** [comment]
+1 @xinlian12 @kushagraThapar 

What is the main reason for hiding setters in GatewayConnectionConfig?

**Comment by kushagraThapar** [comment]
@liquidvapour @adelamodwala @okohub - currently we ask our customers to use Reactor Core's `timeout` API to provide a timeout shorter than 5 seconds. 

https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Flux.html#timeout-java.time.Duration-

**Pull request merged**